src/svgProvider.ts

```diff
@@ -86,4 +86,107 @@ export class SvgFileContentProvider extends SvgDocumentContentProvider {
         let text = fileText ? fileText : '';
         return super.snippet(text);
     }
+}
+
+export class NewSvgDocumentContentProvider {
+
+    protected snippet(properties): string {
+        let showTransGrid = vscode.workspace.getConfiguration('svgviewer').get('transparencygrid');
+        let transparencycolor = vscode.workspace.getConfiguration('svgviewer').get('transparencycolor');
+        let transparencyGridCss = '';
+        if (showTransGrid) {
+            if (transparencycolor != null && transparencycolor !== "") {
+                transparencyGridCss = `
+<style type="text/css">
+.svgbg img {
+    background: `+ transparencycolor + `;
+}
+</style>`;
+            } else {
+                transparencyGridCss = `
+<style type="text/css">
+.svgbg img {
+    background:initial;
+    background-image: url(data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAoCAYAAACM/rhtAAAAeUlEQVRYR+3XMQ4AIQhEUTiU9+/hUGy9Wk2G8luDIS8EMWdmYvF09+JtEUmBpieCJiA96AIiiKAswEsik10JCCIoCrAsiGBPOIK2YFWt/knOOW5Nv/ykQNMTQRMwEERQFWAOqmJ3PIIIigIMahHs3ahZt0xCetAEjA99oc8dGNmnIAAAAABJRU5ErkJggg==);
+    background-position: left,top;
+}
+</style>`;
+            }
+        }
+        return `<!DOCTYPE html><html><head>${transparencyGridCss}</head><body><div class="svgbg"><img src="data:image/svg+xml,${encodeURIComponent(properties)}"></div></body></html>`;
+    }
+
+    public async provideTextDocumentContent(sourceUri: vscode.Uri): Promise<string> {
+        const document = await vscode.workspace.openTextDocument(sourceUri);
+        return this.snippet(document.getText);
```

Comment:
@cssho 
Do not you forget the parentheses?
`return this.snippet(document.getText());`